lib/pure/times.nim

```diff
@@ -1039,6 +1039,123 @@ proc parse*(value, layout: string): TimeInfo =
   info.weekday = getLocalTime(timeInfoToTime(info)).weekday
   return info
 
+# Leap year calculations are adapted from:
+# from http://www.codeproject.com/Articles/7358/Ultra-fast-Algorithms-for-Working-with-Leap-Years
+# The dayOfTheWeek procs are adapated from:
+# from http://stason.org/TULARC/society/calendars/2-5-What-day-of-the-week-was-2-August-1953.html
+
+# Note: for leap years, start date is assumed to be 1 AD.
+# counts the number of leap years up to January 1st of a given year.
+# Keep in mind that if specified year is a leap year, the leap day
+# has not happened before January 1st of that year.
+proc countLeapYears(yearSpan: int): int =
+  (((yearSpan - 1) / 4) - ((yearSpan - 1) / 100) + ((yearSpan - 1)/400)).int
+
+proc countDays(yearSpan: int): int =
+  (yearSpan - 1) * 365 + countLeapYears(yearSpan)
+
+# counts the number of years spanned by a given number of days.
+proc countYears(daySpan: int): int =
+  ((daySpan - countLeapYears(daySpan div 365)) div 365)
+
+proc countYearsAndDays(daySpan: int): tuple[years: int, days: int] =
+  let days = daySpan - countLeapYears(daySpan div 365)
+  result.years = days div 365
+  result.days = days mod 365
+
+type
+  SecondScale* = enum ssMinute, ssHour, ssDay, ssMonth, ssYear
+
+# these are very approximate beyond day
+const secondsIn*: array[SecondScale.low..SecondScale.high, int] =[
+  60,             # minute
+  60*60,          # hour
+  60*60*24,       # day
+  60*60*24*30,    # month (estimate)
+  60*60*24*30*12] # year  (estimate)
+
+const
+  epochStartYear = 1970
+  leapYearsSinceEpoch* = countLeapYears(epochStartYear)
+
+proc dayOfWeek*(day, month, year: int): WeekDay =
+  # This is for the Gregorian calendar
+  # Day & month start from one.
```

Comment:
The docs need to say what this procedure actually does. Same for the procedure below.